cmake/Modules/DeprecationHelpers.cmake

```diff
@@ -0,0 +1,194 @@
+  endif()
+
+  if(DEFINED BROWSER_PANEL_SUPPORT_ENABLED)
+    set(ENABLE_BROWSER_PANELS "${BROWSER_PANEL_SUPPORT_ENABLED}" CACHE BOOL "" FORCE)
+  endif()
+
+  if(DEFINED BUILD_BROWSER)
+    set(ENABLE_BROWSER_SOURCE "${BUILD_BROWSER}" CACHE BOOL "" FORCE)
+  endif()
+
+  if(DEFINED BUILD_CA_ENCODER)
+    set(ENABLE_COREAUDIO_ENCODER "${BUILD_CA_ENCODER}" CACHE BOOL "" FORCE)
+  endif()
+
+  if(DEFINED BUILD_VST)
+    set(ENABLE_VST "${BUILD_VST}" CACHE BOOL "" FORCE)
+  endif()
+
+  if(DEFINED CHECK_FOR_SERVICE_UPDATES)
+    set(ENABLE_SERVICE_UPDATES "${CHECK_FOR_SERVICE_UPDATES}" CACHE BOOL "" FORCE)
+  endif()
+
+  if(DEFINED DEBUG_FFMPEG_MUX)
+    set(ENABLE_FFMPEG_MUX_DEBUG "${DEBUG_FFMPEG_MUX}" CACHE BOOL "" FORCE)
+  endif()
+
+  if(DEFINED DISABLE_IVCAM)
+    if(DISABLE_IVCAM)
+      set(ENABLE_IVCAM OFF CACHE BOOL "" FORCE)
+    else()
+      set(ENABLE_IVCAM ON CACHE BOOL "" FORCE)
+    endif()
+  endif()
+
+  if(DEFINED DISABLE_PLUGINS)
+    if(DISABLE_PLUGINS)
+      set(ENABLE_PLUGINS OFF CACHE BOOL "" FORCE)
+    else()
+      set(ENABLE_PLUGINS ON CACHE BOOL "" FORCE)
+    endif()
+  endif()
+
+  if(DEFINED DISABLE_PYTHON)
+    if(DISABLE_PYTHON)
+      set(ENABLE_SCRIPTING_PYTHON OFF CACHE BOOL "" FORCE)
+    else()
+      set(ENABLE_SCRIPTING_PYTHON ON CACHE BOOL "" FORCE)
+    endif()
+  endif()
+
+  if(DEFINED DISABLE_LUA)
+    if(DISABLE_LUA)
+      set(ENABLE_SCRIPTING_LUA OFF CACHE BOOL "" FORCE)
+    else()
+      set(ENABLE_SCRIPTING_LUA ON CACHE BOOL "" FORCE)
+    endif()
+  endif()
+
+  if(DEFINED DISABLE_SPEEXDSP)
+    if(DISABLE_SPEEXDSP)
+      set(ENABLE_SPEEXDSP OFF CACHE BOOL "" FORCE)
+    else()
+      set(ENABLE_SPEEXDSP ON CACHE BOOL "" FORCE)
+    endif()
+  endif()
+
+  if(DEFINED DISABLE_UPDATE_MODULE)
+    if(DISABLE_UPDATE_MODULE)
+      set(ENABLE_UPDATER OFF CACHE BOOL "" FORCE)
+    else()
+      set(ENABLE_UPDATER ON CACHE BOOL "" FORCE)
+    endif()
+  endif()
+
+  if(DEFINED SHARED_TEXTURE_SUPPORT_ENABLED)
+    set(ENABLE_BROWSER_SHARED_TEXTURE "${SHARED_TEXTURE_SUPPORT_ENABLED}" CACHE BOOL "" FORCE)
+  endif()
+
+  if(DEFINED STATIC_MBEDTLS)
+    set(ENABLE_STATIC_MBEDTLS "${STATIC_MBEDTLS}" CACHE BOOL "" FORCE)
+  endif()
+
+  if(DEFINED UNIX_STRUCTURE AND UNIX_STRUCTURE)
+    set(LINUX_PORTABLE OFF CACHE BOOL "" FORCE)
+  endif()
+
+  if(DEFINED USE_QT_LOOP)
+    set(ENABLE_BROWSER_QT_LOOP "${USE_QT_LOOP}" CACHE BOOL "" FORCE)
+  endif()
+
+  if(DEFINED WITH_RTMPS)
+    set(ENABLE_RTMPS "${WITH_RTMPS}" CACHE STRING "" FORCE)
+  endif()
+endfunction()
+
+function(install_obs_plugin_with_data)
+  message(
+    DEPRECATION
+      "OBS: The install_obs_plugin_with_data command is deprecated and will be removed soon - use 'setup_plugin_target' instead."
+  )
+  _install_obs_plugin_with_data(${ARGV})
+endfunction()
+
+function(install_obs_plugin)
+  message(
+    DEPRECATION
+      "OBS: The install_obs_plugin command is deprecated and will be removed soon - use 'setup_plugintarget' instead."
```

Comment:
Here, the message says to use `setup_plugintarget`.  Above, it says `setup_plugin_target`.  Is one of these a typo?

I'd remove the hyphen/dash and just use separate sentences:
> OBS: The install_obs_plugin command is deprecated and will be removed soon. Use 'setup_plugin_target' instead.

Reply:
Fixed, phrasing adapted.